crates/boundless-market/src/contracts/boundless_market.rs

```diff
@@ -1023,55 +1150,63 @@ impl<P: Provider> BoundlessMarketService<P> {
         lower_bound: Option<u64>,
         upper_bound: Option<u64>,
     ) -> Result<ProofDelivered, MarketError> {
-        let mut upper_block = upper_bound.unwrap_or(self.get_latest_block_number().await?);
-        let start_block = lower_bound.unwrap_or(upper_block.saturating_sub(
-            self.event_query_config.block_range * self.event_query_config.max_iterations,
-        ));
+        let config = self.event_query_config.clone();
+        self.retry_query(&config, || async {
+            let mut upper_block = upper_bound.unwrap_or(self.get_latest_block_number().await?);
+            let initial_upper_block = upper_block;
+            let start_block = lower_bound.unwrap_or(upper_block.saturating_sub(
+                self.event_query_config.block_range * self.event_query_config.max_iterations,
+            ));
```

Comment:
So just a thought, but I wonder if the better long-term path for these is to do an `eth_call` for `requestIsFulfilled(..)` over the block range, and then just query logs through `eth_getLogs` once we know the range. Given the logs call is >2x more expensive from a CU perspective than an `eth_call` for a provider like alchemy and you can rule out ranges larger than the event query range, it seems like it would be cheaper and quicker.

The other main benefit to doing this is that for the problem we are solving for, we would be able to more easily detect, and then just search the range we missed (instead of backsearching two weeks of events per retry). Would also allow us to retry more frequently without worrying about costs or delays.

Reply:
I had a similar thought around using the request info to restrict the search range. Generally trying to not make major changes before the break, so I captured the idea in https://linear.app/boundlessnetwork/issue/BM-2152/fix-sdk-footgun-around-event-query-block-length

I'll add your  idea there too